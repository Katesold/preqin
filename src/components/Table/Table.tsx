import {
  StyledTable,
  TableBody,
  TableHead,
  TableHeader,
  TableHeaderRow,
} from "./styles";

import React from "react";
import { FirmData } from "../../types";
import TableRow from "../TableRow/TableRow";

interface Props {
  firms: FirmData[];
}

const Table: React.FC<Props> = ({ firms }) => {
  return (
    <StyledTable>
      <TableHead>
        <TableHeaderRow>
          <TableHeader>Firm ID</TableHeader>
          <TableHeader>Firm Name</TableHeader>
          <TableHeader>Type</TableHeader>
          <TableHeader>Date Added</TableHeader>
          <TableHeader>Address</TableHeader>
          <TableHeader>Action</TableHeader>
        </TableHeaderRow>
      </TableHead>
      <TableBody>
        {firms.map((firm) => (
          <TableRow key={firm.firmID} data={firm} />
        ))}
      </TableBody>
    </StyledTable>
  );
};

export default Table;
