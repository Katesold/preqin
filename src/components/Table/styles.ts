import styled from "styled-components";

export const StyledTableContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 20px;
`;

export const StyledTable = styled.table`
  width: 100%;
  margin-bottom: 10px;
  border-collapse: collapse;
  text-align: center;
`;

export const TableHead = styled.thead`
  width: 100%;
  background-color: #abfdeb;
`;

export const TableBody = styled.tbody<{ readOnly?: boolean }>`
  width: 100%;
  tr:hover {
    background-color: ${({ readOnly }) => (readOnly ? "none" : "#abfdeb")};
  }
`;

export const TableHeader = styled.th`
  width: 10%;
  height: 28px;
  font-size: 11px;
  line-height: 18px;
  border-top: 1pt solid #e2e2e2;
  color: #757575;
`;

export const TableHeaderRow = styled.tr`
  width: 100%;
  border-bottom: 1pt solid #e2e2e2;
`;
