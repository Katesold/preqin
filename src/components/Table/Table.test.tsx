import { render, screen } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";

import Table from "./Table";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
}));

const data = [
  {
    firmID: "2670",
    firmName: "1199SEIU National Benefit Fund",
    region: "North America",
    address: "498 Seventh Avenue",
    city: "New York",
    stateCounty: "NY",
    zipCode: "10018",
    country: "US",
    website: "http://www.1199seiubenefits.org",
    email: "pension@1199nbf.org",
    tel: "+1 646 473 9200",
    fax: "",
    generalConsultant: "",
    localLanguageFirmName: "",
    secondaryLocations: "",
    firmType: "Private Sector Pension Fund",
    yearEst: "",
    investorCurrency: "USD",
    matchingFunds: "413",
    aumCurrMn: "16260.00",
    aumUSDMn: "16260.00",
    aumEURMn: "15309.42",
    allocationAlternativesPct: "47.44",
    allocationAlternativesCurrMn: "7714.00",
    allocationAlternativesUSDMn: "7714.00",
    allocationAlternativesEURMn: "7041.64",
    allocationEquitiesPct: "30.75",
    allocationEquitiesCurrMn: "5000.00",
    allocationEquitiesUSDMn: "5000.00",
    allocationEquitiesEURMn: "4564.19",
    allocationFixedIncomePct: "14.95",
    allocationFixedIncomeCurrMn: "2431.00",
    allocationFixedIncomeUSDMn: "2431.00",
    allocationFixedIncomeEURMn: "2219.11",
    allocationCashPct: "2.87",
    allocationCashCurrMn: "466.00",
    allocationCashUSDMn: "466.00",
    allocationCashEURMn: "425.38",
    allocationOtherPct: "3.22",
    allocationOtherCurrMn: "523.00",
    allocationOtherUSDMn: "523.00",
    allocationOtherEURMn: "477.41",
    targetAllocationAlternativesPctMin: "",
    about:
      "1199SEIU National Benefit Fund is a labor-management fund that provides comprehensive benefits to working and retired healthcare industry workers, 1199SEIU United Healthcare Workers East members, and their families. The pension fund provides the infrastructure to support additional benefit funds that provide coverage for healthcare workers throughout the industry in hospitals, nursing homes, and home care agencies. It primarily benefits its members through three pension plans: the 1199SEIU Home Care Employees Pension Plan, the 1199SEIU Health Care Employees Pension Fund, and the 1199SEIU Greater New York Pension Fund.\r\n\r\n1199SEIU Health Care Employees Pension Fund, the successor of the 1199 National Pension Fund, was established in 1991. The plan is a noncontributory, multi-employer-defined benefit plan that provides financial security for 1199SEIU retirees.\r\n\r\n1199SEIU Home Care Employees Pension Fund, established in 1997, and 1199SEIU Greater New York Pension Fund, established in 1963, are also noncontributory, multi-employer defined benefit plans.",
  },
];

const routes = [
  {
    path: "/",
    element: <Table firms={data} />,
  },
];

const router = createMemoryRouter(routes, {
  initialEntries: ["/"],
  initialIndex: 0,
});

describe("Table", () => {
  it("should match snapshot", () => {
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Private Sector Pension Fund")).toBeInTheDocument();
  });
});
