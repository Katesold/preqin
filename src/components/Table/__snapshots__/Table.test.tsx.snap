// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot 1`] = `
<div>
  <table
    class="sc-guDLey bZjJOl"
  >
    <thead
      class="sc-dmyCSP eroDKu"
    >
      <tr
        class="sc-jTQCzO kKsiDq"
      >
        <th
          class="sc-eDLKkx cmjbyZ"
        >
          Firm ID
        </th>
        <th
          class="sc-eDLKkx cmjbyZ"
        >
          Firm Name
        </th>
        <th
          class="sc-eDLKkx cmjbyZ"
        >
          Type
        </th>
        <th
          class="sc-eDLKkx cmjbyZ"
        >
          Date Added
        </th>
        <th
          class="sc-eDLKkx cmjbyZ"
        >
          Address
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-hLQSwg lcBNFJ"
    >
      <tr
        class="sc-gLLuof bZixAE"
      >
        <td
          class="sc-iBdnpw klwsqO"
        >
          2670
        </td>
        <td
          class="sc-iBdnpw klwsqO"
        >
          1199SEIU National Benefit Fund
        </td>
        <td
          class="sc-iBdnpw klwsqO"
        >
          Private Sector Pension Fund
        </td>
        <td
          class="sc-iBdnpw klwsqO"
        >
          -
        </td>
        <td
          class="sc-iBdnpw klwsqO"
        >
          498 Seventh Avenue
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
